backend/routes/usernameRoute.js

```diff
@@ -19,7 +20,7 @@ const checkUsernameExists = async (req, res, next) => {
     }
 }
 
-usernameRouter.put('/update-user', checkUsernameExists, async (req, res) => {
+usernameRouter.put('/update-user', authenticateSession, checkUsernameExists, async (req, res) => {
```

Comment:
the route handler should be extracted into a different file or function and you can just do what you are doing with the middleware and call it by its name.